packages/flutter/lib/src/cassowary/constraint.dart

```diff
@@ -5,21 +5,46 @@
 import 'priority.dart';
 import 'expression.dart';
 
+/// A constraint is created by specifying a relationship between two
+/// expressions. The solver tries to satisfy this relationship after the
+/// constraint has been added to it at a set priority.
 enum Relation {
+  /// The relationship between the left and right hand sides of the expression
+  /// is `==`, (lhs == rhs).
   equalTo,
+  /// The relationship between the left and right hand sides of the expression
+  /// is `<=`, (lhs <= rhs).
   lessThanOrEqualTo,
+  /// The relationship between the left and right hand sides of the expression
+  /// is `>=`, (lhs => rhs).
   greaterThanOrEqualTo,
 }
 
+/// A constraint is a relationship between two expressions that the solver tries
+/// to hold true. In case of ambiguities, the solver will use priorities to
+/// determine constraint precedence. Once a constraint is added to the solver,
+/// this priority may not be changed.
```

Comment:
Try to avoid "may not" since it's not clear if it means "might not", "should not", or "cannot".